### ISSUE TITLE
B027 and typing.overload

### ISSUE THREAD

**Issue by yakMM** [PyCQA/flake8-bugbear#308]
Consider the following:

```py
class Foo(ABC):
    @t.overload
    def bar(self, arg: t.Literal[False]) -> t.Literal[False]:
        ...

    @t.overload
    def bar(self, arg: t.Literal[True]) -> t.Literal[True]:
        ...

    def bar(self, arg: bool) -> bool:
        return arg

    @abstractmethod
    def abstract_method(self) -> None:
        ...

```

The overloads of `bar` are considered as "real" methods and B027 warning is emitted:
```
 B027 bar is an empty method in an abstract base class, but has no abstract decorator. Consider adding @abstractmethod.
 B027 bar is an empty method in an abstract base class, but has no abstract decorator. Consider adding @abstractmethod.
```

Is it worth not emitting the warning when typing.overload is used or is it a case that should be supressed manually with `# noqa`?

**Comment by jakkdl** [comment]
B027 checks for `@overload` and `@typing.overload` https://github.com/PyCQA/flake8-bugbear/blob/main/bugbear.py#L672

but since you did `import typing as t` it's not picked up by the linter. Searching github it seems that's relatively common, so I can add that to the logic as well.

**Comment by yakMM** [comment]
Ah okay makes sense, I didn't know this kind of details mattered, I thought the imports were evaluated

**Comment by jakkdl** [comment]
Typecheckers evaluate imports, but AST checkers like flake8 usually don't try and just looks at identifiers hoping/assuming they're well-named.

Anyway, fixed with #309 - forgot to mention this issue in it.

**Comment by cooperlees** [comment]
So, is this fine to be closed now?

**Comment by jakkdl** [comment]
yep!
